controllers/stories.js

```diff
@@ -0,0 +1,45 @@
+const storyManager = require('../public/javascripts/data/storyManager');
+const paragraph = require('../public/javascripts/data/paragraph');
+
+exports.getAllStories = function(req, res, next) {
+
+    let currentUserName = res.locals.userName;
+    res.render('index', { stories: storyManager.displayAllStories(), currentUserName: currentUserName});
+};
+
+exports.addStory = function(req, res, next) {
+
+    // Idea: send user to the add to story page, so they can start the first paragraph of their new story.
+
+    let storyTitle = req.body.addStoryInput;
+    let userId = req.session.user['id'];
+    
+    storyManager.addStory(storyTitle, userId);
+    res.redirect('/');
+};
+
+exports.viewStory = function(req, res, next) {
+
+    let currentUserName = res.locals.userName;
+    let storyId = req.body['storyId'];
+
+    res.render('story', { story: storyManager.getStory(storyId), currentUserName: currentUserName });
+};
+
+// POST request /addparagraph from form on /addparagraph
+exports.addParagraph = function(req, res, next) {
+    
+    // Do a check here to see if there is a paragraph to submit
+    // This /story POST is shared between the button on the home page to view a story and submitting a paragraph on /story
+    if(!req.body.paragraph_input) {
+        next();
```

Comment:
Is this a validation check? Do you need to return some view data indicating to the user that the paragraph input form was empty and that is why their paragraph was not added?

Reply:
Here I'm sharing two POST requests on  /stories routing. They both have the same middleware  except this one function
` stories.addParagraph`. So it's just a quick check to see if I can bypass that function (if they're on the home page and clicking the button to view the story (/story POST)

This is the route each of the two POST requests get sent to
```
// view story button POST request on "/ homepage" and submit paragraph POST request on "/story"
router.post('/story', handleAuth.requireLogin, handleAuth.currentUser, stories.addParagraph, stories.viewStory);
```

Reply:
I gotcha. For your view story route, `/story`, I'd recommend using a GET. You can pass the Id to the router and with your link. For instance:

index.ejs:
```html
<a href="/stories/<%= story.id %>"
```

router.js
```js
router.get('/story/:id', handleAuth.requireLogin, handleAuth.currentUser, stories.viewStory)
```

Reply:
Great advice. I was able to implement this.

`exports.viewStory = function(req, res, next) {
    
    let currentUserName = req.session.user['user_name'];
    
    // GET request
    if(req.method == "GET") {

       let storyId =  req.query['storyId'];
       res.render(`story`, { story: storyManager.getStory(storyId), currentUserName: currentUserName });
    }
    // POST request
    else {
        
        let storyId = req.body['storyId'];
        res.render('story', { story: storyManager.getStory(storyId), currentUserName: currentUserName });
    };
};`